src/PrestaShopBundle/Translation/DataCollectorTranslator.php

```diff
@@ -29,7 +29,23 @@
 
 use Symfony\Component\Translation\DataCollectorTranslator as BaseTranslator;
 
-class DataCollectorTranslator extends BaseTranslator
+class DataCollectorTranslator extends BaseTranslator implements TranslatorInterface
 {
     use PrestaShopTranslatorTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isLanguageLoaded($locale)
+    {
+        return $this->__call('isLanguageLoaded', [$locale]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clearLanguage($locale)
+    {
+        return  $this->__call('clearLanguage', [$locale]);
```

Comment:
I'm not sure about these two methods. They seem to use the [base translator's __call() method](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Translation/DataCollectorTranslator.php#L113-L116) but it needs a decorated translator.

Why don't you call `$this->translator->clearLanguage($locale)`, since the decorated translator should anyway be an instance of Prestashop's one?
You could even type-hint this in the class with `/** @​property PrestashopTranslator $translator */` (of course you would need to import the namespace for auto-completion in IDEs).

Reply:
I had forgotten about this, the answer is pretty simple: the translator property is private on the original DataCollectorTranslator, so I can't access it 😢 

That said I think I could have used `parent::isLanguageLoaded()` and it would have been routed automagically all the same. But I don't know if it would have been clearer that way either anyway 😁

Reply:
Calling `parent::isLanguageLoaded()` is definitely the way to go. Relying on magic methods is almost never a good option. Especially when the base method is reachable through `parent::` 🙂

Reply:
I had to revert to this because PHPStan doesn't support calls to `parent` if the method is not declared  in the parent and routed with `__call()` ☹️ 

See https://github.com/PrestaShop/PrestaShop/pull/18082/checks?check_run_id=1375456805

Reply:
@eternoendless PHPStan is not a true god that you have to follow! If you're certain that you have the right class at the right moment, and you tested the feature, then it works, and you can add an exception to PHPStan. If you do so, remember to add a small comment to explain why there's an exception :)